File: services/scout-go/pkg/messages/messages.go
package messages

import "time"

type Target struct {
	ID           string    `json:"id"`
	BountyType   string    `json:"bountyType,omitempty"`
	Kind         string    `json:"kind"`
	ChainID      int       `json:"chainId,omitempty"`
	Address      string    `json:"address,omitempty"`
	Repo         string    `json:"repo,omitempty"`
	CommitSha    string    `json:"commitSha,omitempty"`
	SourceURL    string    `json:"sourceUrl,omitempty"`
	TxHash       string    `json:"txHash,omitempty"`
	DiscoveredAt time.Time `json:"discoveredAt"`
	Priority     float64   `json:"priority"`
	TVLUsd       float64   `json:"tvlUsd,omitempty"`
}

type Severity string

const (
	SevInfo     Severity = "info"
	SevLow      Severity = "low"
	SevMedium   Severity = "medium"
	SevHigh     Severity = "high"
	SevCritical Severity = "critical"
)

type Finding struct {
	ID          string   `json:"id"`
	TargetID    string   `json:"targetId"`
	BountyType  string   `json:"bountyType,omitempty"`
	Category    string   `json:"category"`
	Severity    Severity `json:"severity"`
	Tools       []string `json:"tools"`
	Location    Location `json:"location,omitempty"`
	Description string   `json:"description"`
}

type Location struct {
	File      string `json:"file"`
	LineStart int    `json:"lineStart"`
	LineEnd   int    `json:"lineEnd"`
}

type VerifiedExploit struct {
	ID                    string    `json:"id"`
	FindingID             string    `json:"findingId"`
	PoCPath               string    `json:"pocPath"`
	ImpactType            string    `json:"impactType"`
	ValidatedAgainstConfig bool     `json:"validatedAgainstConfig"`
	AttackerModel         string    `json:"attackerModel"`
	ModifiedValidator     bool      `json:"modifiedValidator"`
	Justification         string    `json:"justification,omitempty"`
	VerifiedAt            time.Time `json:"verifiedAt"`
}

type Disclosure struct {
	ID          string    `json:"id"`
	ExploitID   string    `json:"exploitId"`
	Lane        string    `json:"lane"`
	X402URL     string    `json:"x402Url,omitempty"`
	TxHash      string    `json:"txHash,omitempty"`
	StorageHash string    `json:"storageHash,omitempty"`
	PublishedAt time.Time `json:"publishedAt"`
}

File: services/auditor-rs/src/messages.rs
use serde::{Deserialize, Serialize};
use chrono::{DateTime, Utc};

#[derive(Debug, Serialize, Deserialize, Clone)]
#[serde(rename_all = "lowercase")]
pub enum TargetKind { Onchain, Github, Immunefi }

#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct Target {
    pub id: String,
    pub kind: TargetKind,
    #[serde(rename = "chainId", skip_serializing_if = "Option::is_none")]
    pub chain_id: Option<u64>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub address: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub repo: Option<String>,
    #[serde(rename = "commitSha", skip_serializing_if = "Option::is_none")]
    pub commit_sha: Option<String>,
    #[serde(rename = "sourceUrl", skip_serializing_if = "Option::is_none")]
    pub source_url: Option<String>,
    #[serde(rename = "discoveredAt")]
    pub discovered_at: DateTime<Utc>,
    pub priority: f64,
    #[serde(rename = "tvlUsd", skip_serializing_if = "Option::is_none")]
    pub tvl_usd: Option<f64>,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
#[serde(rename_all = "lowercase")]
pub enum Severity { Info, Low, Medium, High, Critical }

#[derive(Debug, Serialize, Deserialize, Clone, Default)]
pub struct Location {
    pub file: String,
    #[serde(rename = "lineStart")]
    pub line_start: u32,
    #[serde(rename = "lineEnd")]
    pub line_end: u32,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct Finding {
    pub id: String,
    #[serde(rename = "targetId")]
    pub target_id: String,
    pub category: String,
    pub severity: Severity,
    pub tools: Vec<String>,
    #[serde(default)]
    pub location: Location,
    #[serde(default)]
    pub description: String,
}
